subprojects/core/src/main/java/org/gradle/internal/vfs/WindowsFileWatcherRegistry.java

```diff
@@ -37,11 +36,11 @@
 
     public WindowsFileWatcherRegistry(Set<Path> watchRoots, ChangeHandler handler) {
         super(
-            Native.get(WindowsFileEventFunctions.class).startWatching(
+            callback -> Native.get(WindowsFileEventFunctions.class).startWatching(
                 watchRoots.stream()
                     .map(Path::toString)
                     .collect(Collectors.toList()),
-                (type, path) -> WatcherEvent.createEvent(type, path).dispatch(handler)
+                callback
```

Comment:
This kind of constructor callback still bothers me, but I have no better idea atm.